packages/storage-ui/src/Components/Modules/FileSystemItem/itemOperations.tsx

```diff
@@ -0,0 +1,152 @@
+import React from "react"
+import {
+  IMenuItem,
+  EditSvg,
+  DeleteSvg,
+  DownloadSvg,
+  ExportSvg,
+  ShareAltSvg,
+  RecoverSvg,
+  ZoomInSvg,
+  EyeSvg,
+  ExclamationCircleInverseSvg
+} from "@chainsafe/common-components"
+import { Trans } from "@lingui/macro"
+import { FileOperation } from "../../../Contexts/types"
+import { ISelectedFile } from "../../../Contexts/FileBrowserContext"
+import { FileSystemItem } from "../../../Contexts/StorageContext"
+
+export function getItemMenuOptions(params: {
+  menuIconClass: string
+  file: FileSystemItem
+  editFile?: (item: ISelectedFile) => void
+  deleteFile?: (item: ISelectedFile) => void
+  downloadFile?: (item: ISelectedFile) => void
+  moveFile?: (item: ISelectedFile) => void
+  previewFile?: (file: FileSystemItem) => void
+  viewFolder?: (item: ISelectedFile) => void
+  showFileInfo?: (item: ISelectedFile) => void
+  itemOperations: FileOperation[]
+}
+) {
+  const {
+    menuIconClass,
+    file,
+    viewFolder,
+    editFile,
+    deleteFile,
+    downloadFile,
+    moveFile,
+    previewFile,
+    showFileInfo,
+    itemOperations
+  } = params
```

Comment:
Nit, making the interface explicit and restructuring the params right away.
```suggestion

interface MenuOptionsParams {
  menuIconClass: string
  file: FileSystemItem
  inSharedFolder: boolean
  itemOperations: FileOperation[]
  editFile?: (item: FileSystemItem) => void
  deleteFile?: (item: FileSystemItem) => void
  downloadFile?: (item: FileSystemItem) => void
  moveFile?: (item: FileSystemItem) => void
  handleShare?: (item: FileSystemItem) => void
  recoverFile?: (item: FileSystemItem) => void
  previewFile?: (item: FileSystemItem) => void
  showFileInfo?: (item: FileSystemItem) => void
  reportFile?: (item: FileSystemItem) => void
  viewFolder?: (item: FileSystemItem) => void
}

export const getItemMenuOptions = ({
  menuIconClass,
  file,
  inSharedFolder,
  itemOperations,
  editFile,
  deleteFile,
  downloadFile,
  moveFile,
  handleShare,
  recoverFile,
  previewFile,
  showFileInfo,
  reportFile,
  viewFolder
}: MenuOptionsParams) => {
```